x-pack/plugins/security_solution_ess/kibana.jsonc

```diff
@@ -9,7 +9,10 @@
     "browser": true,
     "configPath": ["xpack", "securitySolutionEss"],
     "requiredPlugins": [
+      "actions",
       "securitySolution",
+      "stackConnectors",
+      "triggersActionsUi",
```

Comment:
Do you know if these dependencies are needed? I don't see them being used in the plugin.

Reply:
Those transitive dependencies were required because `x-pack/plugins/security_solution_ess/public/upselling/pages/attack_discovery/index.tsx` included `UpgradeButtons` from the `@kbn/elastic-assistant` package:

```typescript
 import { UpgradeButtons } from '@kbn/elastic-assistant';
```

, which depended on:

```json
      "actions",
      "stackConnectors",
      "triggersActionsUi",
```

and therefore generated the following errors without the above entries in `x-pack/plugins/security_solution_ess/kibana.jsonc`:

```
ERROR in kibana/x-pack/packages/kbn-elastic-assistant/impl/connectorland/use_load_action_types/index.tsx
       │          Module not found: Error: import [@kbn/actions-plugin/common] references a public export of the [actions] bundle, but that bundle is not in the "requiredPlugins" or "requiredBundles" list in the plugin manifest [x-pack/plugins/security_solution_ess/kibana.jsonc]

ERROR in x-pack/packages/kbn-elastic-assistant/impl/connectorland/connector_button/index.tsx
       │          Module not found: Error: import [@kbn/stack-connectors-plugin/public/common] references a public export of the [stackConnectors] bundle, but that bundle is not in the "requiredPlugins" or "requiredBundles" list in the plugin manifest [x-pack/plugins/security_solution_ess/kibana.jsonc]


ERROR in x-pack/packages/kbn-elastic-assistant/impl/connectorland/use_load_connectors/index.tsx
       │          Module not found: Error: import [@kbn/triggers-actions-ui-plugin/public/common/constants] references a public export of the [triggersActionsUi] bundle, but that bundle is not in the "requiredPlugins" or "requiredBundles" list in the plugin manifest [x-pack/plugins/security_solution_ess/kibana.jsonc]
```

To eliminate the dependency on `@kbn/elastic-assistant` (and therefore the transitive dependencies), a new component to replace it was added to `x-pack/plugins/security_solution_ess/public/upselling/pages/attack_discovery/upgrade/upgrade_buttons.tsx`, and the dependency on `@kbn/elastic-assistant` was removed in [this commit](https://github.com/elastic/kibana/pull/188788/commits/648313c7880da774559decb6c2a64ede65614ff6)